pyccel/parser/syntax/openmp.py

```diff
@@ -31,142 +35,547 @@ def __init__(self, **kwargs):
         """
         self.stmt = kwargs.pop('stmt')
 
-        super(OpenmpStmt, self).__init__(**kwargs)
+        super().__init__(**kwargs)
 
     @property
     def expr(self):
         if DEBUG:
             print("> OpenmpStmt: expr")
 
         stmt = self.stmt
-        if isinstance(stmt, OmpEndClause):
-            return stmt.expr
-        elif isinstance(stmt, OmpParallelConstruct):
-            return stmt.expr
-        elif isinstance(stmt, OmpLoopConstruct):
-            return stmt.expr
-        elif isinstance(stmt, OmpSingleConstruct):
+        if isinstance(stmt, (OmpEndClause, OmpParallelConstruct, OmpLoopConstruct,
+                             OmpSingleConstruct, OmpCriticalConstruct, OmpBarrierConstruct,
+                             OmpMasterConstruct, OmpMaskedConstruct, OmpTaskLoopConstruct,
+                             OmpSimdConstruct, OmpAtomicConstruct, OmpTaskWaitConstruct,
+                             OmpTaskyieldConstruct, OmpTaskConstruct, OmpFlushConstruct,
+                             OmpCancelConstruct, OmpTargetConstruct, OmpTeamsConstruct,
+                             OmpDistributeConstruct, OmpSectionConstruct, OmpSectionsConstruct)):
```

Comment:
Is this all the `OmpAnnotatedComment`s?

Reply:
no those are a `textx` rules, here we still didn't create an `OmpAnnotatedComments`

Reply:
Ah oui, pardon